cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -700,6 +706,10 @@ private boolean getPartitionLoad(HttpServletRequest request, HttpServletResponse
                               "CPU (%)", "DISK (MB)", "NW_IN (KB/s)", "NW_OUT (KB/s)")
                       .getBytes(StandardCharsets.UTF_8));
       for (Partition p : sortedPartitions) {
+        if ((topicToFilter != null && !p.topicPartition().topic().equals(topicToFilter)) ||
```

Comment:
I am wondering if `TOPIC_PARAM` is required if `PARTITION_PARAM` is set.
I don't think there is value in just fetching partition 1 for all topics for example. Since the logic topic view is top level to partitions and similarly `TOPIC_PARAM` is required with `PARITION_PARAM`.

Reply:
in a scenario that client forget to set `TOPIC_PARAM`  and just `PARTITION_PARAM`, I think returning the corresponding partitions for all topic is ok since client  will realize they missed the `TOPIC_PARAM` and either resend the request with the parameter added or they can filter out the information in this response, which is probably of the superset of what they want .

Reply:
It is better to return an error saying the topic is a required field when partition param is passed. In Kafka, the topic is the higher level abstraction which is used everywhere and we don't really just give the partition number without the topic anywhere.